### PR TITLE
doc/user: add v0.26.0 release notes

### PR DESCRIPTION
**Pull request by benesch**
### Motivation

* Release notes PR.

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

File: doc/user/content/release-notes.md [hunk 1]
```diff
@@ -19,11 +19,44 @@ the PR description instead. They will be migrated here during the release
 process by the release notes team.
 {{< /comment >}}
 
+{{% version-header version=v0.26.0 lts=true %}}
+
+v0.26.0 is a **long-term support (LTS)** release of Materialize. We will support
+the v0.26 series through at least (TODO WHEN?), issuing patch releases as
```
Comment (morsapaes) [inline doc/user/content/release-notes.md:25]: Hanging `TODO`.

File: doc/user/content/sql/create-source/kafka.md [hunk 2]
```diff
@@ -138,9 +138,10 @@ Note that:
 
 - The `DEBEZIUM` envelope is incompatible with this option.
 
-#### Partition, offset, timestamp
+#### Partition, offset, timestamp, headers
```
Comment (morsapaes) [inline doc/user/content/sql/create-source/kafka.md:141]: Is this all the documentation there is on header metadata? Sigh. I'm on mobile for a bit, but will have a look later to improve it if it is (not blocking for the notes, though the anchor link might change).

**Review by morsapaes — CHANGES_REQUESTED** [review]
There's as sneaky `TODO` in there, otherwise LGTM! Sorry I didn't get to them before you this time. 👹

File: doc/user/content/release-notes.md (hunk 1 above)
Comment (benesch) [reply doc/user/content/release-notes.md]: Updated now!

File: doc/user/content/sql/create-source/kafka.md (hunk 2 above)
Comment (morsapaes) [reply doc/user/content/sql/create-source/kafka.md:141]: Wanna pull this out? I have a PR on the side for the headers docs.

**Review by morsapaes — APPROVED** [review]

File: doc/user/content/sql/create-source/kafka.md (hunk 2 above)
Comment (morsapaes) [reply doc/user/content/sql/create-source/kafka.md:141]: https://github.com/MaterializeInc/materialize/pull/11768

Comment (benesch) [reply doc/user/content/sql/create-source/kafka.md:141]: Will do! I'm holding off on merging this until tomorrow afternoon, so I'll just rebase after you merge #11768 and update the link appropriately.

Comment (benesch) [reply doc/user/content/sql/create-source/kafka.md]: Updated as promised!

**Comment by benesch** [comment]
I think this is ready to go now. I'll merge after our all hands meeting tomorrow.

**Pull request merged**